In [1]:
import pandas as pd

# Data import and cleaning

## Exploring WVS time series 5.0

In [2]:
df = pd.read_csv('../WVS_time_series/WVS_Time_Series_1981-2022_v5_0.csv')

In [4]:
df

version                        doi  S001  S002VS  S003  \
0       5-0-0 (2024-04-30)  doi.org/10.14281/18241.25     2       3     8   
1       5-0-0 (2024-04-30)  doi.org/10.14281/18241.25     2       3     8   
2       5-0-0 (2024-04-30)  doi.org/10.14281/18241.25     2       3     8   
3       5-0-0 (2024-04-30)  doi.org/10.14281/18241.25     2       3     8   
4       5-0-0 (2024-04-30)  doi.org/10.14281/18241.25     2       3     8   
...                    ...                        ...   ...     ...   ...   
443483  5-0-0 (2024-04-30)  doi.org/10.14281/18241.25     2       7   716   
443484  5-0-0 (2024-04-30)  doi.org/10.14281/18241.25     2       7   716   
443485  5-0-0 (2024-04-30)  doi.org/10.14281/18241.25     2       7   716   
443486  5-0-0 (2024-04-30)  doi.org/10.14281/18241.25     2       7   716   
443487  5-0-0 (2024-04-30)  doi.org/10.14281/18241.25     2       7   716   

       COUNTRY_ALPHA  COW_NUM COW_ALPHA  S004       S006  ...     Y023A  \
0                ALB      339       ALB     1          1  ...  0.000000   
1                ALB      339       ALB     1          2  ...  0.111111   
2                ALB      339       ALB     1          3  ...  0.000000   
3                ALB      339       ALB     1          4  ...  0.000000   
4                ALB      339       ALB     1          5  ...  0.000000   
...              ...      ...       ...   ...        ...  ...       ...   
443483           ZWE      552       ZIM    -4  716071211  ...  0.000000   
443484           ZWE      552       ZIM    -4  716071212  ...  0.000000   
443485           ZWE      552       ZIM    -4  716071213  ...  0.000000   
443486           ZWE      552       ZIM    -4  716071214  ...  0.000000   
443487           ZWE      552       ZIM    -4  716071215  ...  0.000000   

           Y023B     Y023C  Y024A  Y024B  Y024C  survself  tradrat5   TradAgg  \
0       0.444444  0.444444   0.33    0.0  0.165 -1.011276  0.160847  0.158964   
1       0.444444  0.444444   0.33    0.0  0.165 -0.688773  0.409283  0.558946   
2       0.444444  0.444444   0.33    0.5  0.415       NaN       NaN       NaN   
3       0.333333  0.333333   0.33    0.0  0.165 -1.319863  0.733956  1.081669   
4       0.333333  0.333333   0.00    0.5  0.250       NaN       NaN       NaN   
...          ...       ...    ...    ...    ...       ...       ...       ...   
443483  0.000000  0.000000   0.66    0.5  0.580 -0.738923 -0.774148 -1.346378   
443484  0.000000  0.777778   1.00    0.0  0.500 -0.445039  0.282275  0.354463   
443485  0.000000  0.000000   0.33    0.0  0.165 -0.925464 -0.436454 -0.802691   
443486  0.000000  0.000000   0.00    0.0  0.000 -2.398929  1.218936  1.862486   
443487  0.000000  0.000000   0.00    0.0  0.000 -1.633334  0.249829  0.302224   

        SurvSAgg  
0      -1.792410  
1      -1.208679  
2            NaN  
3      -2.350952  
4            NaN  
...          ...  
443483 -1.299450  
443484 -0.767520  
443485 -1.637090  
443486 -4.304062  
443487 -2.918335  

[443488 rows x 1046 columns]

In [13]:
df[['S003', 'COUNTRY_ALPHA']].info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 443488 entries, 0 to 443487
Data columns (total 2 columns):
 #   Column         Non-Null Count   Dtype 
---  ------         --------------   ----- 
 0   S003           443488 non-null  int64 
 1   COUNTRY_ALPHA  443488 non-null  object
dtypes: int64(1), object(1)
memory usage: 6.8+ MB


All of the country values are complete - so we can filter safely only for the United States values.

In [16]:
df['S002VS'].value_counts()

S002VS
7    97220
6    89565
5    83975
3    77818
4    60045
2    24558
1    10307
Name: count, dtype: int64

In [24]:
df_first2waves = df[(df['S002VS'] == 1) | (df['S002VS'] == 2)]

for column in ['S002VS', 'S003', 'COUNTRY_ALPHA']:
    print(df_first2waves[column].value_counts())

S002VS
2    24558
1    10307
Name: count, dtype: int64
S003
710    4332
484    3368
356    2500
410    2221
392    2215
32     2007
643    1961
76     1782
724    1510
152    1500
348    1464
756    1400
36     1228
792    1030
112    1015
246    1003
566    1001
156    1000
616     938
203     924
703     466
Name: count, dtype: int64
COUNTRY_ALPHA
ZAF    4332
MEX    3368
IND    2500
KOR    2221
JPN    2215
ARG    2007
RUS    1961
BRA    1782
ESP    1510
CHL    1500
HUN    1464
CHE    1400
AUS    1228
TUR    1030
BLR    1015
FIN    1003
NGA    1001
CHN    1000
POL     938
CZE     924
SVK     466
Name: count, dtype: int64


In [27]:
df_first2waves[['S002VS', 'S003', 'COUNTRY_ALPHA']].info()

<class 'pandas.core.frame.DataFrame'>
Index: 34865 entries, 4006 to 425816
Data columns (total 3 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   S002VS         34865 non-null  int64 
 1   S003           34865 non-null  int64 
 2   COUNTRY_ALPHA  34865 non-null  object
dtypes: int64(2), object(1)
memory usage: 1.1+ MB


All the waves are included! But why is there no information for the United States, when theoretically there should also be questions for waves 1 and 2?

In [14]:
df_us = df[df['COUNTRY_ALPHA'] == 'USA']

df_us

version                        doi  S001  S002VS  S003  \
401631  5-0-0 (2024-04-30)  doi.org/10.14281/18241.25     2       3   840   
401632  5-0-0 (2024-04-30)  doi.org/10.14281/18241.25     2       3   840   
401633  5-0-0 (2024-04-30)  doi.org/10.14281/18241.25     2       3   840   
401634  5-0-0 (2024-04-30)  doi.org/10.14281/18241.25     2       3   840   
401635  5-0-0 (2024-04-30)  doi.org/10.14281/18241.25     2       3   840   
...                    ...                        ...   ...     ...   ...   
410445  5-0-0 (2024-04-30)  doi.org/10.14281/18241.25     2       7   840   
410446  5-0-0 (2024-04-30)  doi.org/10.14281/18241.25     2       7   840   
410447  5-0-0 (2024-04-30)  doi.org/10.14281/18241.25     2       7   840   
410448  5-0-0 (2024-04-30)  doi.org/10.14281/18241.25     2       7   840   
410449  5-0-0 (2024-04-30)  doi.org/10.14281/18241.25     2       7   840   

       COUNTRY_ALPHA  COW_NUM COW_ALPHA  S004       S006  ...     Y023A  \
401631           USA        2       USA     1          1  ...  0.000000   
401632           USA        2       USA     1          2  ...  0.555556   
401633           USA        2       USA     1          3  ...  1.000000   
401634           USA        2       USA     1          4  ...  0.000000   
401635           USA        2       USA     1          5  ...  0.000000   
...              ...      ...       ...   ...        ...  ...       ...   
410445           USA        2       USA    -4  840072715  ...  0.666667   
410446           USA        2       USA    -4  840071538  ...  0.888889   
410447           USA        2       USA    -4  840071446  ...  0.444444   
410448           USA        2       USA    -4  840071167  ...  0.555556   
410449           USA        2       USA    -4  840072217  ...  0.555556   

           Y023B     Y023C  Y024A  Y024B  Y024C  survself  tradrat5   TradAgg  \
401631  0.000000  0.444444   0.00    0.0  0.000  0.557763 -1.243387 -2.101853   
401632  0.666667  0.444444   0.66    1.0  0.830  2.080791 -0.781275 -1.357852   
401633  1.000000  1.000000   1.00    0.0  0.500       NaN       NaN       NaN   
401634  0.000000  0.000000   0.33    0.5  0.415  0.956679 -1.568910 -2.625945   
401635  1.000000  1.000000   1.00    0.0  0.500  1.020697 -0.169883 -0.373511   
...          ...       ...    ...    ...    ...       ...       ...       ...   
410445  0.555556  0.555556   0.00    0.0  0.000       NaN       NaN       NaN   
410446  0.777778  0.777778   1.00    1.0  1.000  0.312523  2.282574  3.574944   
410447  0.555556  0.555556   1.00    1.0  1.000  0.999833 -0.738978 -1.289755   
410448  0.000000  0.777778    NaN    NaN    NaN       NaN       NaN       NaN   
410449  0.000000  0.777778   0.00    0.5  0.250  1.049813 -1.685598 -2.813813   

        SurvSAgg  
401631  1.047551  
401632  3.804231  
401633       NaN  
401634  1.769589  
401635  1.885462  
...          ...  
410445       NaN  
410446  0.603666  
410447  1.847697  
410448       NaN  
410449  1.938161  

[8819 rows x 1046 columns]

In [15]:
df_us['S002VS'].value_counts()

S002VS
7    2596
6    2232
3    1542
5    1249
4    1200
Name: count, dtype: int64

Note how there is no data for the first 2 waves. This is solved below by merging the latest version of the dataset (above) with a previous version, which includes the data for the US.

## Missing US data for the first 2 waves

In [ ]:
df2020 = pd.read_csv('../WVS_TimeSeries_1981_2020_ascii_v2_0.csv')

df2020

S001  S002VS  S003 COUNTRY_ALPHA  COW_NUM COW_ALPHA  S006        S007  \
0          2       1    32           ARG      160       ARG     1   320120001   
1          2       1    32           ARG      160       ARG     2   320120002   
2          2       1    32           ARG      160       ARG     3   320120003   
3          2       1    32           ARG      160       ARG     4   320120004   
4          2       1    32           ARG      160       ARG     5   320120005   
...      ...     ...   ...           ...      ...       ...   ...         ...   
432477     2       7   716           ZWE      552       ZIM  1211  7160721211   
432478     2       7   716           ZWE      552       ZIM  1212  7160721212   
432479     2       7   716           ZWE      552       ZIM  1213  7160721213   
432480     2       7   716           ZWE      552       ZIM  1214  7160721214   
432481     2       7   716           ZWE      552       ZIM  1215  7160721215   

        S008  S010  ...  Y022B  Y022C     Y023A     Y023B     Y023C  Y024A  \
0         -4  -4.0  ...    NaN    NaN  0.000000  0.000000  0.000000   0.00   
1         -4  -4.0  ...    NaN    NaN  0.000000  0.000000  0.000000   0.00   
2         -4  -4.0  ...    NaN    NaN  0.111111  0.333333  0.555556   0.00   
3         -4  -4.0  ...    NaN    NaN  0.333333  0.555556  0.666667   1.00   
4         -4  -4.0  ...    NaN    NaN  0.444444  0.222222  0.333333   0.00   
...      ...   ...  ...    ...    ...       ...       ...       ...    ...   
432477    50  77.0  ...   0.00   0.66  0.000000  0.000000  0.000000   0.66   
432478    50  80.0  ...   0.00   1.00  0.000000  0.000000  0.777778   1.00   
432479    60  59.0  ...   0.66   0.66  0.000000  0.000000  0.000000   0.33   
432480    60  83.0  ...   0.66   0.66  0.000000  0.000000  0.000000   0.00   
432481    60  86.0  ...   0.66   0.66  0.000000  0.000000  0.000000   0.00   

        Y024B  Y024C   TradAgg  SurvSAgg  
0         NaN    NaN -1.358583 -1.124587  
1         NaN    NaN -0.142223 -2.673256  
2         NaN    NaN  0.084509 -2.033451  
3         NaN    NaN  3.529956 -0.442163  
4         NaN    NaN  1.913126 -1.911873  
...       ...    ...       ...       ...  
432477    0.5  0.580 -1.365047 -1.280893  
432478    0.0  0.500  0.315010 -0.756891  
432479    0.0  0.165 -0.844314 -1.613268  
432480    0.0  0.000  1.775634 -4.353077  
432481    0.0  0.000  0.236233 -2.928095  

[432482 rows x 1072 columns]

In [ ]:
df_first2waves = df2020[(df2020['S002VS'] == 1) | (df2020['S002VS'] == 2)]

for column in ['S002VS', 'S003', 'COUNTRY_ALPHA']:
    print(df_first2waves[column].value_counts())

S002VS
2    28127
1    14840
Name: count, dtype: int64
S003
710    4332
840    4164
484    3368
124    2984
356    2500
410    2221
392    2215
32     2007
643    1961
76     1782
724    1510
152    1500
348    1464
756    1400
36     1228
792    1030
112    1015
246    1003
566    1001
156    1000
752     954
616     938
203     924
703     466
Name: count, dtype: int64
COUNTRY_ALPHA
ZAF    4332
USA    4164
MEX    3368
CAN    2984
IND    2500
KOR    2221
JPN    2215
ARG    2007
RUS    1961
BRA    1782
ESP    1510
CHL    1500
HUN    1464
CHE    1400
AUS    1228
TUR    1030
BLR    1015
FIN    1003
NGA    1001
CHN    1000
SWE     954
POL     938
CZE     924
SVK     466
Name: count, dtype: int64


This includes data for the US! Now we will only keep those observations of the US, to concatenate afterwards the datasets.

In [39]:
df_us_missing = df_first2waves[df_first2waves['COUNTRY_ALPHA'] == 'USA']

df_us_missing

S001  S002VS  S003 COUNTRY_ALPHA  COW_NUM COW_ALPHA  S006        S007  \
12515     1       1   840           USA        2       USA     1  8400120001   
12516     1       1   840           USA        2       USA     2  8400120002   
12517     1       1   840           USA        2       USA     3  8400120003   
12518     1       1   840           USA        2       USA     4  8400120004   
12519     1       1   840           USA        2       USA     5  8400120005   
...     ...     ...   ...           ...      ...       ...   ...         ...   
42962     1       2   840           USA        2       USA  1835  8400221835   
42963     1       2   840           USA        2       USA  1836  8400221836   
42964     1       2   840           USA        2       USA  1837  8400221837   
42965     1       2   840           USA        2       USA  1838  8400221838   
42966     1       2   840           USA        2       USA  1839  8400221839   

       S008  S010  ...  Y022B  Y022C     Y023A     Y023B     Y023C  Y024A  \
12515    -4  -4.0  ...    NaN    NaN  0.555556  0.444444  0.444444    NaN   
12516    -4  -4.0  ...    NaN    NaN  0.444444  0.444444  0.444444    NaN   
12517    -4  -4.0  ...    NaN    NaN       NaN       NaN       NaN    NaN   
12518    -4  -4.0  ...    NaN    NaN  0.000000  0.000000  0.000000    NaN   
12519    -4  -4.0  ...    NaN    NaN  0.000000  0.444444  0.666667    NaN   
...     ...   ...  ...    ...    ...       ...       ...       ...    ...   
42962    -4  -4.0  ...    NaN    NaN  0.000000  0.555556  0.555556   0.66   
42963    -4  -4.0  ...    NaN    NaN  0.000000  0.000000  0.000000   0.33   
42964    -4  -4.0  ...    NaN    NaN  0.444444  0.666667  0.666667   1.00   
42965    -4  -4.0  ...    NaN    NaN  0.000000  0.555556  0.555556    NaN   
42966    -4  -4.0  ...    NaN    NaN  0.444444  1.000000  1.000000   0.33   

       Y024B  Y024C   TradAgg  SurvSAgg  
12515    NaN    NaN       NaN       NaN  
12516    NaN    NaN       NaN       NaN  
12517    NaN    NaN       NaN       NaN  
12518    NaN    NaN       NaN       NaN  
12519    NaN    NaN       NaN       NaN  
...      ...    ...       ...       ...  
42962    NaN    NaN       NaN       NaN  
42963    0.5  0.415       NaN       NaN  
42964    0.5  0.750 -0.229636  4.338274  
42965    0.0    NaN       NaN       NaN  
42966    1.0  0.665  1.751926  2.976972  

[4164 rows x 1072 columns]

Check the columns that are different in the 5.0 dataset and the 2.0 dataset.

In [41]:
diff_columns = set(df_us.columns).symmetric_difference(df_us_missing.columns)
print(len(diff_columns))
print(diff_columns)

64
{'F144_01', 'G040', 'G022S', 'X034', 'X006_02', 'X003R2', 'doi', 'E155', 'W010', 'W007', 'E158', 'S004', 'A007', 'X028_01', 'C001_01', 'W004', 'X003R', 'W011', 'X025CS', 'E069_18A', 'X046', 'G061', 'F025_01old', 'D043_01', 'E161', 'E147', 'G043', 'G036', 'E154', 'S019', 'C002_01', 'E069_16', 'S016', 'E159', 'G033', 'G038', 'W005_2', 'survself', 'G041', 'W008', 'X007_02', 'E162', 'MODE', 'W009', 'X002_03', 'E157', 'E156', 'X032R', 'E181_EVS5', 'W001', 'F025_WVS', 'G034', 'E069_67', 'A080_01', 'E160', 'F025old', 'tradrat5', 'A080_02', 'version', 'G035', 'E179_WVS7LOC', 'F030', 'F025', 'D066_01'}


Maybe all of these different columns should be dropped. For now, though, we will concatenate the two datasets.

## Merging the data

In [ ]:
df_us_concat = pd.concat([df_us, df_us_missing])

df_us_concat

version                        doi  S001  S002VS  S003  \
401631  5-0-0 (2024-04-30)  doi.org/10.14281/18241.25     2       3   840   
401632  5-0-0 (2024-04-30)  doi.org/10.14281/18241.25     2       3   840   
401633  5-0-0 (2024-04-30)  doi.org/10.14281/18241.25     2       3   840   
401634  5-0-0 (2024-04-30)  doi.org/10.14281/18241.25     2       3   840   
401635  5-0-0 (2024-04-30)  doi.org/10.14281/18241.25     2       3   840   
...                    ...                        ...   ...     ...   ...   
42962                  NaN                        NaN     1       2   840   
42963                  NaN                        NaN     1       2   840   
42964                  NaN                        NaN     1       2   840   
42965                  NaN                        NaN     1       2   840   
42966                  NaN                        NaN     1       2   840   

       COUNTRY_ALPHA  COW_NUM COW_ALPHA  S004  S006  ...  W010  W011  X002_03  \
401631           USA        2       USA   1.0     1  ...   NaN   NaN      NaN   
401632           USA        2       USA   1.0     2  ...   NaN   NaN      NaN   
401633           USA        2       USA   1.0     3  ...   NaN   NaN      NaN   
401634           USA        2       USA   1.0     4  ...   NaN   NaN      NaN   
401635           USA        2       USA   1.0     5  ...   NaN   NaN      NaN   
...              ...      ...       ...   ...   ...  ...   ...   ...      ...   
42962            USA        2       USA   NaN  1835  ...  -4.0  -4.0     -4.0   
42963            USA        2       USA   NaN  1836  ...  -4.0  -4.0     -4.0   
42964            USA        2       USA   NaN  1837  ...  -4.0  -4.0     -4.0   
42965            USA        2       USA   NaN  1838  ...  -4.0  -4.0     -4.0   
42966            USA        2       USA   NaN  1839  ...  -4.0  -4.0     -4.0   

        X006_02  X007_02  X025CS  X028_01  X032R  X034  X046  
401631      NaN      NaN     NaN      NaN    NaN   NaN   NaN  
401632      NaN      NaN     NaN      NaN    NaN   NaN   NaN  
401633      NaN      NaN     NaN      NaN    NaN   NaN   NaN  
401634      NaN      NaN     NaN      NaN    NaN   NaN   NaN  
401635      NaN      NaN     NaN      NaN    NaN   NaN   NaN  
...         ...      ...     ...      ...    ...   ...   ...  
42962      -4.0     -4.0    -4.0     -4.0   -4.0  -4.0  -2.0  
42963      -4.0     -4.0    -4.0     -4.0   -4.0  -4.0   4.0  
42964      -4.0     -4.0    -4.0     -4.0   -4.0  -4.0   1.0  
42965      -4.0     -4.0    -4.0     -4.0   -4.0  -4.0   3.0  
42966      -4.0     -4.0    -4.0     -4.0   -4.0  -4.0   1.0  

[12983 rows x 1091 columns]

Now, all of the waves are complete for the US!

Useful variables:
- `S002VS`: survey wave. 1 1981-1984; 2 1989-1993; 3 1994-1998; 4 1999-2004; 5 2005-2009; 6 2010-2014; 7 2017-2022
- `S003`: country code ISO 3166 (840 for the US).
- `S007`: unified respondent number. Therefore, **each row in the dataset is a different respondent!**
- `S012`: Date of interview YYYYMMDD.
- Weight variables that should be used for aggregating the data by wave (read more about it [here](https://www.worldvaluessurvey.org/WVSContents.jsp?CMSID=WEIGHT&CMSID=WEIGHT)):
  - `S017`: Weight assigned to the respondent. Weight by country.
  - `S018`: Equilibrated weight-1000. According to the reference above, we should use this to consistently aggregate the data per wave and year! In summary, "Each respondent has a weight to ensure national representativeness. Always calculate weighted means or proportions so that your country-level aggregates reflect the population accurately."
- `S020`: Year of survey YYYY.

Useful questions:
- **A (society and personal view)**: A001-A006 (importance in life of various topics), A008 (feeling of happiness), A013 (feeling of loneliness), A046-A049 (abortion), A046 (politics discussion with friends), A124_01-61 (preferences, for neighbors), A170 (overall satisfaction with life), A200-A202 (position of people of different ages in society), A208 (respect towards older people in society).
- **B (environment)**: B001-B004, B007 (can be used as proxies for the willingness to take action to tackle climate change and have better sus with the environment), B021-B023 (concern about climate change, biodiversity loss and pollution).****
- **C (employment)**: C006 (satisfaction with financial situation of HH), C033 (job satisfaction). Continue with C040.